marker-api.js

```diff
@@ -1,780 +1,780 @@
-            },
-
-            // Project (filter) only necessary fields
-            { "$project" : {
-				"createdDate" : 1,
-				"tags" : 1,
-				"photo_hash" : 1,
-				"geometry" : 1,
-				"user_info._id" : 1,
-				"approved": 1
-	    	}}
-        ])
-        .exec(function (err, markers) {
-            if (err)
-                return def.reject({'message': 'An internal error occurred', 'status': 500});
-
-            if (!markers || !markers[0])
-                return def.reject({'status': 404, message: 'No markers found with ids: ' + obj_ids});
-
-            // Build return data
-            returnData = markers;
-            returnData.forEach(function (marker) {
-
-                // Make joined username field a property of main object
-                if (marker.user_info && marker.user_info.length) {
-                    marker.user_id = marker.user_info[0]._id;
-                }
-                delete marker.user_info;
-            });
-
-            def.resolve(returnData);
-        });
-
-    return def.promise;
-}
-
-/**
- * Get base 64 image data for a photo by it's id an size
- * @param markers {object} key ->
- */
-function get_photo_b64 (markers) {
-    var def = Q.defer(),
-        sizes,
-        size_suffix,
-        b64_data = {},
-        reads = [],
-        ids;
-
-        console.log('photo requests: ' + JSON.stringify(markers));
+    console.log('validation complete');
 
-    if (!markers) {
-        console.log('invalid data for get_photo_b64');
-        return def.reject({'status': 500, 'message': 'An internal error occurred'});
-    }
+      // Format coords for geometry storage
+      lat = parseFloat(fields.latitude);
+      lng = parseFloat(fields.longitude);
+      marker.geometry = {
+          "type": "Point",
+          "coordinates": [lng, lat]
+      };
 
-    ids = Object.keys(markers);
+      // Split tags on spaces
+      marker.tags = fields.tags.split(' ');
 
-    ids.forEach((id) => {
-        console.log('getting img data for ' + id);
+      marker.user_id = req.user._id;
 
-        sizes = markers[id];
+      marker.photo_hash = files.photo.hash;
 
-        // Return data
-        b64_data[id] = {};
+      // Convert created date string as Date object 
+      marker.createdDate = new Date(fields.created);
 
-        // Make array if size is single string
-        if (!Array.isArray(sizes)) {
-            sizes = [sizes];
-        }
+      // save the marker 
+      marker.save(function(err) {
 
-        // Loop over each size for this marker
-        sizes.forEach((size) => {
+          if (err) {
+              console.log('Error in saving marker: ' + err);
 
-            // Make and store promise for async read
-            var read = Q.defer();
-            reads.push(read.promise);
+              console.log('err code: ' + err.code);
+              if (err.code === 16755) {
+                  return res.status(422).json({message: 'Latitude and longitude are not geographically valid.'});
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/charlieblevins/dukweb/pullRequest?prid=9134160)
```suggestion
                  return res.status(422).json({message: "Latitude and longitude are not geographically valid."});
```